rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -962,6 +962,86 @@ public void updateCollections(Transaction rskTx) throws IOException {
+
+        BtcTransaction svpFundTransactionUnsigned = createSvpFundTransactionWithoutSignatures(proposedFederation);
+        provider.setSvpFundTxHashUnsigned(svpFundTransactionUnsigned.getHash());
+        performPegoutActions(svpFundTransactionUnsigned, rskTx.getHash(), getSvpFundTxSentAmount(proposedFederation, svpFundTransactionUnsigned.getOutputs()));
+    }
+
+    protected BtcTransaction createSvpFundTransactionWithoutSignatures(Federation proposedFederation) throws InsufficientMoneyException {
+        Coin spendableValueFromProposedFederation = bridgeConstants.getSpendableValueFromProposedFederation();
+        Coin feePerKb = feePerKbSupport.getFeePerKb();
+        Wallet activeFederationWallet = getActiveFederationWallet(true);
+
+        BtcTransaction svpFundTransactionUnsigned = new BtcTransaction(bridgeConstants.getBtcParams());
+        svpFundTransactionUnsigned.setVersion(BTC_TX_VERSION_2);
+
+        svpFundTransactionUnsigned.addOutput(spendableValueFromProposedFederation, proposedFederation.getAddress());
+        // complete tx with input and change output
+        SendRequest sendRequest = completeSvpSpendTransactionRequestToSend(svpFundTransactionUnsigned, feePerKb);
+        activeFederationWallet.completeTx(sendRequest);
+
+        return svpFundTransactionUnsigned;
+    }
+
+    private SendRequest completeSvpSpendTransactionRequestToSend(BtcTransaction transaction, Coin feePerKb) {
+        SendRequest sendRequest = SendRequest.forTx(transaction);
+        sendRequest.changeAddress = getActiveFederationAddress();
+        sendRequest.feePerKb = feePerKb;
+        sendRequest.missingSigsMode = Wallet.MissingSigsMode.USE_OP_ZERO;
+        sendRequest.recipientsPayFees = false;
+
+        return sendRequest;
+    }
+
+    private static Coin getSvpFundTxSentAmount(Federation proposedFederation, List<TransactionOutput> pegoutTransactionOutputs) {
+        // pegout value "requested" from proposed federation is the value actually sent
+        // since the active federation is paying for the fees
+
+        Script proposedFederationScriptPubKey = proposedFederation.getP2SHScript();
+        Optional<TransactionOutput> outputToProposedFederationOpt = pegoutTransactionOutputs.stream()
+            .filter(output -> output.getScriptPubKey().equals(proposedFederationScriptPubKey))
+            .findFirst();
+
+        if (!outputToProposedFederationOpt.isPresent()) {
+            String message = "The output for the proposed federation should be present in the svp fund transaction.";
+            logger.warn(message);
+            throw new IllegalStateException(message);
+        }
+        TransactionOutput outputToProposedFederation = outputToProposedFederationOpt.get();
+
+        return outputToProposedFederation.getValue();
+    }
+
+    private void performPegoutActions(BtcTransaction pegoutTransaction, Keccak256 rskTxHash, Coin requestedPegoutAmount) throws IOException {
```

Comment:
```suggestion
    private void confirmReleaseRequest(BtcTransaction pegoutTransaction, Keccak256 releaseCreationTxHash, Coin requestedAmount) throws IOException {
```
Tried to give a more meaningful name to the method and params. The idea is that this can be used for migration, pegouts, and fundTx right?